src/main/java/com/sumologic/log4j/http/SumoHttpSender.java

```diff
@@ -25,136 +25,121 @@
-
-    private volatile String url = null;
-    private volatile ProxySettings proxySettings = null;
-
-    private int connectionTimeout = 1000;
-    private int socketTimeout = 60000;
-    private volatile CloseableHttpClient httpClient = null;
-
-    public ProxySettings getProxySettings() {
-        return proxySettings;
-    }
-
-    public void setProxySettings(ProxySettings proxySettings) {
-        this.proxySettings = proxySettings;
-    }
-
-    public void setRetryInterval(long retryInterval) {
-        this.retryInterval = retryInterval;
-    }
-
-    public void setUrl(String url) {
-        this.url = url;
-    }
-
-    public void setConnectionTimeout(int connectionTimeout) {
-        this.connectionTimeout = connectionTimeout;
-    }
-
-    public void setSocketTimeout(int socketTimeout) {
-        this.socketTimeout = socketTimeout;
-    }
 
-    public boolean isInitialized() {
-        return httpClient != null;
-    }
-
-    public void init() {
-        RequestConfig requestConfig = RequestConfig.custom()
-                .setSocketTimeout(socketTimeout)
-                .setConnectTimeout(connectionTimeout)
-                .build();
-
-        HttpClientBuilder builder = HttpClients.custom()
-                .setConnectionManager(new PoolingHttpClientConnectionManager())
-                .setDefaultRequestConfig(requestConfig);
-
-        HttpProxySettingsCreator creator = new HttpProxySettingsCreator(proxySettings);
-        creator.configureProxySettings(builder);
-
-        httpClient = builder.build();
-    }
-
-    public void close() throws IOException {
-        httpClient.close();
-        httpClient = null;
-    }
-
-    public void send(String body, String name) {
-        keepTrying(body, name);
-    }
-
-    private void keepTrying(String body, String name) {
-        boolean success = false;
-        do {
-            try {
-                trySend(body, name);
-                success = true;
-            } catch (Exception e) {
-                try {
-                    Thread.sleep(retryInterval);
-                } catch (InterruptedException e1) {
-                    break;
-                }
-            }
-        } while (!success && !Thread.currentThread().isInterrupted());
-    }
-
-    private void trySend(String body, String name) throws IOException {
-        HttpPost post = null;
+public class SumoHttpSender
+{
+  private static final Logger logger = StatusLogger.getLogger();
+
+  private final long retryInterval;
+  private final String url;
+  private final CloseableHttpClient httpClient;
+  private final String name;
+  private final String host;
+  private final String category;
+
+  public SumoHttpSender(
+      long retryInterval,
+      String url,
+      CloseableHttpClient httpClient,
+      String name,
+      String host,
+      String category
+  )
+  {
+    this.retryInterval = retryInterval;
+    this.url = url;
+    this.httpClient = httpClient;
```

Comment:
It would be nice to use Netty instead of Apache HTTP client. Not as part of this PR though.